include/git2/errors.h

```diff
@@ -31,6 +31,8 @@ typedef enum {
 	 * GIT_EUSER is a special error that is never generated by libgit2
 	 * code.  You can return it from a callback (e.g to stop an iteration)
 	 * to know that it was generated by the callback and not by libgit2.
+	 *
+	 * You may also use error codes below GIT_EMIN for the same purpose.
```

Comment:
We generally don't suggest that end-users use our exit codes from callbacks.  Obviously, there's nothing _stopping_ them, but our general rule is that a user callback should return `GIT_EUSER` to cancel the function.

Reply:
libgit2 only uses a very small amount of error codes - I think it might be beneficial for applications to be able to return other codes than just GIT_EUSER. After all, a callback may have many different reasons for triggering an abort. It is only a suggestion. I'm not particularly insistent on this.

Reply:
They definitely can but we expect them to store that state themselves.  We don’t make any guarantees that - if a user returned a -3 that we wouldn’t try to do something based on that result, thinking that *we* had returned it down in the stack.  This isn’t something that I expect that we do anywhere but the only guarantees we make are around `GIT_EUSER`